.github/workflows/e2e.yml

```
@@ -87,12 +87,17 @@ jobs:
       - uses: pnpm/action-setup@c3b53f6a16e57305370b4ae5a540c2077a1d50dd # pin@v2.2.4
         with:
           version: 7
-      - run: cargo build --bin sui-test-validator --bin sui --profile dev
+      - name: Build devnet validator
+        run: cargo build --bin sui-test-validator --profile dev
 
-      # checkout the current branch
+      # checkout and build the current branch
       - uses: actions/checkout@7dd9e2a3dc350cf687eb1b2a4fadfee8c8e49675 # pin@v3
         with:
           clean: false
+      - name: Build sui bin
+        run:
+          rustup component remove cargo && rustup component add cargo
+          cargo build --bin sui
```

shou
```suggestion
          cargo build --bin sui-test-validator --bin sui --profile dev
```

should we just combine the two?

---

My understanding of this workflow is that it
1. Checks out devnet branch, builds validator
2. Checks out a commit, tests against the commit.

If we rebuild `sui-test-validator`, the test will not be a test for compatibility. Is that intended?

---

Oh you are right